sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Sample.java

```diff
@@ -233,10 +244,10 @@ private SampleAnyCombineFn(long limit) {
       List<T> res = iter.next();
       while (iter.hasNext()) {
```

Comment:
Consider using ImmutableList.copyOf(Iterables.limit(Iterables.concat(accumulators.iterator()), limit));

https://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/Iterables.html

Reply:
I'd like to avoid making a copy of the list here: mergeAccumulators allows modification in-place.

Reply:
I see, sgtm.